api/control/v1alpha1/types.go

```diff
@@ -457,6 +467,39 @@ const (
 // Type checking errors are posted to the status.Errors of the Binding and BindingPolicy.
 type Expression string
 
+// ExpressionContext defines what an Expression can reference regarding the workload object.
+type ExpressionContext struct {
+	// `inventory` holds the inventory record for the workload object.
+	Inventory InventoryRecord `json:"inventory"`
+
+	// `obj` holds a copy of the workload object as read from the WDS.
+	Obj runtime.RawExtension `json:"obj"`
+
+	// `returned` holds the fragment of the workload object that was returned to the core from the WEC.
+	Returned ReturnedState `json:"returned"`
+
+	// `propagation` holds data about the current state of the work on propagating
+	// the object's state from WDS to WEC and from WEC to WDS.
+	Propagation PropagatationData `json:"propagation"`
+}
+
+// InventoryRecord is what appears in the inventory for a given WEC.
+type InventoryRecord struct {
+	// the name of the WEC.
+	Name string `json:"name"`
+}
+
+type ReturnedState struct {
+	Status runtime.RawExtension `json:"status"`
+}
+
+type PropagatationData struct {
+	// `lastReturnedUpdateTimestamp` is the time of the last update to any
+	// of the returned object state in the core.
+	// Before the first such update, this is the zero value of `time.Time`.
+	LastReturnedUpdateTimestamp metav1.Time `json:"lastReturnedUpdateTimestamp"`
```

Comment:
Is `metav1.Time` the best choice here? A previous version of this PR used a type from a protobuf library. I think that we want to optimize for convenience in use in the user's CEL expressions.

Reply:
Looking at https://github.com/google/cel-spec/blob/master/doc/langdef.md#functions, I wonder if `google.protobuf.Timestamp` is a better choice.

Reply:
Also, since this has seconds precision, I suspect that a viable choice would be simply an int32 or uint32 holding seconds since Unix epoch (1970.0).